### PR TITLE
Submit IDE activity

### PR DESCRIPTION
**Pull request by Taikelenn**
# Description of the PR

Submits IDE activity as a file with a magic name "_ideact_result"

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [X] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [X] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [X] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [X] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

**Review by jaakkonarhi — APPROVED** [review]
Looks good to me, just need to make an issue to show the submission payload to the student

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java [hunk 1]
```diff
@@ -301,6 +307,54 @@ private void trySubmit(@NotNull Project project)
     logger.debug("Finished submitting exercise");
   }
 
+  /**
+   * Submits a tutorial.
+   */
+  public void submitTutorial(@NotNull Project project,
+                             @NotNull TutorialViewModel tutorialViewModel) {
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java:314]: I don't think this belongs here. If you want to have it in the same place as `trySubmit`, consider moving them both out of the action.

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/TutorialAction.java [hunk 2]
```diff
@@ -142,22 +142,25 @@ public boolean confirmCancel(@NotNull TutorialViewModel tutorialViewModel) {
...
   }
 
-  private void onTutorialComplete(@NotNull MainViewModel mainViewModel) {
+  private void onTutorialComplete(@NotNull MainViewModel mainViewModel,
+                                  @NotNull Project project) {
     TutorialViewModel viewModel = mainViewModel.tutorialViewModel.get();
     if (viewModel != null) {
       viewModel.getTutorial().tutorialCompleted.removeCallback(mainViewModel);
+      if (viewModel.isCompleted() && dialogs.finishAndSubmit(viewModel)) {
+        new SubmitExerciseAction().submitTutorial(project, viewModel);
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/intellij/actions/TutorialAction.java:159]: This is what I meant in my previous comment. Actions are not designed to be created this way.

File: src/main/resources/resources.properties [hunk 3]
```diff
@@ -201,7 +201,7 @@ ui.BannerView.courseEnded=The course has ended.
 ## Tutorial
 ui.tutorial.TutorialAction.confirmStart=A tutorial will start.
 ui.tutorial.TutorialAction.confirmCancel=Cancel tutorial?
-ui.tutorial.TutorialAction.end=The tutorial has ended.
+ui.tutorial.TutorialAction.end=<html>The tutorial has ended. Would you like to submit your result to the A+ system?<br>If you choose not to submit the result now, you will have to do the tutorial again.</html>
```
Comment (OlliKiljunen) [inline src/main/resources/resources.properties:204]: Ideally, student should be able to choose to send it later but if that's difficult to implement, this goes for now. Let's discuss in the weekly with the stakeholders.

**Review by OlliKiljunen — COMMENTED** [review]
Very nice and concise implementation! 👍 What first seemed like a complicated issue, turned out to have good straightforward solution, finding which, however, required us to get a better understanding what's the issue was all about. Very well done!

However, before approving, I'd like to see some refactoring (see the first review comment of mine), or alternatively, hear good reasoning behind the current design. ;)

**Pull request merged**